### ISSUE TITLE
assignability failure with generic PEP 695 alias as type context

### ISSUE THREAD

**Issue by carljm** [astral-sh/ty#2682]
This emits a bogus error:

> ```py
> type MaybeList[T] = list[T] | T
> 
> def test[X: int](items: list[X]) -> list[X]:
>     a: MaybeList[str | int] = list(items) #
>     return items
> ```
> 
> https://play.ty.dev/1757f2f8-b251-4a94-96d2-51ff3743f6d9 

But this version (with a PEP 613 alias, or a legacy type alias) works:

```py
from typing import TypeVar, TypeAlias

T = TypeVar("T")

MaybeList: TypeAlias = list[T] | T

def test[X: int](items: list[X]) -> list[X]:
    a: MaybeList[str | int] = list(items)
    return items
```

 _Originally posted by @DetachHead in [#2314](https://github.com/astral-sh/ty/issues/2314#issuecomment-3821654048)_
